UnityProject/Assets/Scripts/Objects/Machines/Jukebox.cs

```diff
@@ -1,285 +1,305 @@
-		private float MinSoundDistance = 4;
+	[SerializeField]
+	private float MinSoundDistance = 4;
 
-		[SerializeField]
-		private float MaxSoundDistance = 10;
+	[SerializeField]
+	private float MaxSoundDistance = 10;
 
-		[SerializeField]
-		[Range(0, 1)]
-		private float Volume = 1;
+	[SerializeField]
+	[Range(0, 1)]
+	private float Volume = 1;
 
-		[SerializeField]
-		[Range(0, 360)]
-		private float Spread = 0;
+	[SerializeField]
+	[Range(0, 360)]
+	private float Spread = 0;
 
-		private List<AudioSource> musics;
 
-		private SpriteRenderer spriteRenderer;
+	private SpriteRenderer spriteRenderer;
+	/// <summary>
+	/// The current state of the jukebox powered/overpowered/underpowered/no power
+	/// </summary>
+	private PowerStates CurrentState;
 
-		/// <summary>
-		/// The current state of the jukebox powered/overpowered/underpowered/no power
-		/// </summary>
-		private PowerStates CurrentState;
+	/// <summary>
+	/// The current state of the jukebox powered/overpowered/underpowered/no power
+	/// </summary>
+	private APCPoweredDevice APCConnectionHandler;
 
-		/// <summary>
-		/// The current state of the jukebox powered/overpowered/underpowered/no power
-		/// </summary>
-		private APCPoweredDevice APCConnectionHandler;
+	private Integrity integrity;
+	private APCPoweredDevice power;
+	private RegisterTile registerTile;
+	private int currentSongTrackIndex = 0;
+	private float startPlayTime;
 
-		private Integrity integrity;
-		private APCPoweredDevice power;
-		private RegisterTile registerTile;
-		private int currentSongTrackIndex = 0;
-		private float startPlayTime;
+	private string soundSpawnToken;
 
-		public bool IsPlaying { get; set; } = false;
+	private async Task<AddressableAudioSource> GetAddressableAudioSourceFromCache()
+	{
+		AddressableAudioSource addressableAudioSourceParam = new AddressableAudioSource(SoundManager.Instance.MusicLibrary.ElementAt(currentSongTrackIndex));
+		AddressableAudioSource addressableAudioSourceFromCache = await SoundManager.GetAddressableAudioSourceFromCache(new List<AddressableAudioSource> { addressableAudioSourceParam });
+		return addressableAudioSourceFromCache;
+	}
 
-		public string TrackPosition {
-			get {
-				return $"{currentSongTrackIndex + 1} / {musics.Count}";
-			}
-		}
+	public bool IsPlaying { get; set; } = false;
 
-		public string SongName {
-			get {
-				string songName = musics[currentSongTrackIndex].clip.name;
-				return $"{songName.Split('_')[0]}";
-			}
+	public string TrackPosition
+	{
+		get
+		{
+			return $"{currentSongTrackIndex + 1} / {SoundManager.Instance.MusicLibrary.Count}";
 		}
+	}
 
-		public string Artist {
-			get {
-				string songName = musics[currentSongTrackIndex].clip.name;
-				string artist = songName.Contains("_") ? songName.Split('_')[1] : "Unknown";
-				return $"{artist}";
-			}
-		}
+	public async Task<string> GetSongNameAsync()
+	{
+		AddressableAudioSource addressableAudioSourceFromCache = await GetAddressableAudioSourceFromCache();
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)